### PR TITLE
Gracefully error out if integer overflow in ReaxFF or QEq

### PR DESCRIPTION
**Pull request by stanmoore1**
**Summary**

Gracefully error out if integer overflow in ReaxFF or QEq

**Related Issue(s)**

https://github.com/lammps/lammps/issues/4194

**Author(s)**

Stan Moore (SNL), in discussion with Steve Plimpton

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

Yes

### CODE REVIEW

**Comment by stanmoore1** [comment]
@akohlmey I'm done working on this, it should be ready for review

File: src/QEQ/fix_qeq.cpp [hunk 1]
```diff
@@ -261,7 +262,10 @@ void FixQEq::allocate_matrix()
     i = ilist[ii];
     m += numneigh[i];
   }
-  m_cap = MAX((int)(m * safezone), mincap * MIN_NBRS);
+  bigint m_cap_big = MAX((int)(m * safezone), mincap * MIN_NBRS);
```
Comment (akohlmey) [inline src/QEQ/fix_qeq.cpp]:
```suggestion
  bigint m_cap_big = MAX((bigint)(m * safezone), (bigint)mincap * MIN_NBRS);
```
Shouldn't  there be casts to bigint here?

File: src/REAXFF/fix_qeq_reaxff.cpp [hunk 2]
```diff
@@ -360,7 +361,10 @@ void FixQEqReaxFF::allocate_matrix()
     i = ilist[ii];
     m += numneigh[i];
   }
-  m_cap = MAX((int)(m * safezone), mincap * REAX_MIN_NBRS);
+  bigint m_cap_big = MAX((int)(m * safezone), mincap * REAX_MIN_NBRS);
```
Comment (akohlmey) [inline src/REAXFF/fix_qeq_reaxff.cpp]:
```suggestion
  bigint m_cap_big = MAX((bigint)(m * safezone), (bigint)mincap * REAX_MIN_NBRS);
```
Same as in fix_qeq.cpp

**Comment by stanmoore1** [comment]
@akohlmey good catch, I will make the change

**Comment by stanmoore1** [comment]
I don't think we need the second cast since it is a minimum size, e.g. the default mincap = 50, and the REAX_MIN_NBRS = 100 --> 5000 by default.

**Comment by akohlmey** [comment]
> I don't think we need the second cast since it is a minimum size, e.g. the default mincap = 50, and the REAX_MIN_NBRS = 100 --> 5000 by default.

It doesn't hurt and compilers may throw warnings for comparing different size integers.

**Comment by stanmoore1** [comment]
@akohlmey I tweaked it a bit, I think everything has to get promoted to `double` for the comparison anyway. If you want to change it that is fine with me

**Review by akohlmey — APPROVED** [review]

**Pull request merged**